PlayList.java

```diff
@@ -108,13 +180,30 @@ public void add(PlayList other) {
      *  If start is negative or greater than size - 1, returns -1.
      */
     private int minIndex(int start) {
-        //// replace the following statement with your code
-        return 0;
+    if(start<0||start>size-1)   
+    {   
+    return -1;
+    }
+    int minValue=1000000;
+    int minIndex =maxSize;
+    for(int i=start;i<size;i++)
```

Comment:
How do you know the real min value isn't larger?
```suggestion
    int minValue=tracks[start].getDuration();
    int minIndex =start;
    for(int I=start+1;i<size;i++)
```